app/views/Data/Workspace/index.tsx

```diff
@@ -255,50 +270,64 @@ export default function Workspace(props: Props) {
     *  the query even when the list is empty.
     */
     const context = useMemo(() => ({ additionalTypenames: ['TableType'] }), []);
-    const [tablePreview, setTablePreview] = useState(false);
     const [relationTableId, setRelationTableId] = useState<string>();
-
+    const [tableId, setTableId] = useState<string>();
     const [
         tablesAddedToWorkspaceResult,
     ] = useQuery({
         query: tablesAddedToWorkspaceQueryDocument,
         context,
     });
 
+    const [
+        workspaceTableResult,
+    ] = useQuery({
+        query: workspaceTableQueryDocument,
+        variables: {
+            id: tableId ?? '',
+        },
+    });
```

Comment:
we can move this query to the component where it is used. This will prevent from the query executing on render with `id` as an empty string.